fdbserver/include/fdbserver/workloads/TenantConsistency.actor.h

```diff
@@ -36,222 +36,120 @@
+		ASSERT_EQ(tenantData.tenantMap.size(), tenantData.tenantCount);
 
-				break;
-			} catch (Error& e) {
-				wait(safeThreadFutureToFuture(tr->onError(e)));
+		std::set<int64_t> tenantsInTenantGroupIndex;
+		for (auto const& [groupName, tenants] : tenantData.tenantGroupIndex) {
+			for (auto const& tenant : tenants) {
+				tenantsInTenantGroupIndex.insert(tenant);
 			}
 		}
 
-		ASSERT(!tenantList.more);
-		std::map<int64_t, TenantMapEntryImpl> localMap =
-		    std::map<int64_t, TenantMapEntryImpl>(tenantList.results.begin(), tenantList.results.end());
-
-		ASSERT(!tenantNameIndexList.more);
-		self->metadata.tenantNameIndex =
-		    std::map<TenantName, int64_t>(tenantNameIndexList.results.begin(), tenantNameIndexList.results.end());
-
-		ASSERT(!tenantTombstoneList.more);
-		self->metadata.tenantTombstones =
-		    std::set<int64_t>(tenantTombstoneList.results.begin(), tenantTombstoneList.results.end());
-
-		ASSERT(!tenantGroupList.more);
-		self->metadata.tenantGroupMap =
-		    std::map<TenantGroupName, TenantGroupEntry>(tenantGroupList.results.begin(), tenantGroupList.results.end());
-
-		for (auto t : tenantGroupTenantTuples.results) {
-			ASSERT_EQ(t.size(), 2);
-			TenantGroupName tenantGroupName = t.getString(0);
-			int64_t tenantId = t.getInt(1);
-			ASSERT(self->metadata.tenantGroupMap.count(tenantGroupName));
-			ASSERT(localMap.count(tenantId));
-			self->metadata.tenantGroupIndex[tenantGroupName].insert(tenantId);
-			ASSERT(self->metadata.tenantsInTenantGroupIndex.insert(tenantId).second);
-		}
-		ASSERT_EQ(self->metadata.tenantGroupIndex.size(), self->metadata.tenantGroupMap.size());
-
-		return localMap;
-	}
-
-	void validateTenantMetadata(std::map<int64_t, TenantMapEntry> tenantMap) {
-		ASSERT(metadata.clusterType == ClusterType::METACLUSTER_DATA ||
-		       metadata.clusterType == ClusterType::STANDALONE);
-		ASSERT_LE(tenantMap.size(), CLIENT_KNOBS->MAX_TENANTS_PER_CLUSTER);
-		ASSERT_EQ(tenantMap.size(), metadata.tenantCount);
-		ASSERT_EQ(metadata.tenantNameIndex.size(), metadata.tenantCount);
-
-		for (auto [tenantId, tenantMapEntry] : tenantMap) {
+		for (auto [tenantId, tenantMapEntry] : tenantData.tenantMap) {
 			ASSERT_EQ(tenantId, tenantMapEntry.id);
-			// Only standalone clusters will have lastTenantId set
-			// For Metacluster, the lastTenantId field is updated for MetaclusterMetadata
-			// and not TenantMetadata
-			if (metadata.clusterType != ClusterType::METACLUSTER_DATA) {
-				if (TenantAPI::getTenantIdPrefix(tenantId) == TenantAPI::getTenantIdPrefix(metadata.lastTenantId)) {
-					ASSERT_LE(tenantId, metadata.lastTenantId);
+			ASSERT_EQ(tenantData.tenantNameIndex[tenantMapEntry.tenantName], tenantId);
+
+			// Data clusters do not keep their last tenant ID up to date while part of a metacluster
+			if (tenantData.clusterType != ClusterType::METACLUSTER_DATA) {
+				if (TenantAPI::getTenantIdPrefix(tenantId) == TenantAPI::getTenantIdPrefix(tenantData.lastTenantId)) {
```

Comment:
I think `nextTenantIdPrefixMatches` can do this for us with less clutter

Reply:
True, though it looks like we'd have to work around the error case logging a SevError event in that function. Since it didn't change here, we can tackle that later if desired.